### PR TITLE
docs(pruning): Improve pruning and snapshot recovery docs

### PR DESCRIPTION
**Pull request by slowli**
## What ❔

Improves various pruning and snapshot recovery docs.

## Why ❔

Makes docs more thorough and clearer.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Code has been formatted via `zk fmt` and `zk lint`.

### CODE REVIEW

File: docs/guides/external-node/08_pruning.md [hunk 1]
```diff
@@ -1,13 +1,37 @@
...
+threshold. Data is pruned both from Postgres and from tree (RocksDB). Pruning happens continuously (i.e., does not
+require stopping the node) in the background during normal node operation. It generally does not significantly influence
+node performance.
 
-> [!NOTE]
->
-> If you need a node with data retention period of up to a few days, please set up a node from a
-> [_snapshot_](07_snapshots_recovery.md) and wait for it to have enough data. Pruning an archival node can take
-> unpractical amount of time. In the future we will be offering pre-pruned DB snapshots with a few months of data.
+Types of pruned data in Postgres include:
```
Comment (tomg10) [inline docs/guides/external-node/08_pruning.md:8]: I'd rephrase this so that we don't enumerate all types of entities that are pruned, do you think describing that no data from pruned batches is available is not enough?

File: docs/guides/external-node/08_pruning.md [hunk 2]
```diff
@@ -1,13 +1,37 @@
 # Pruning
 
-It is possible to configure ZKsync Node to periodically remove all data from batches older than a configurable
-threshold. Data is pruned both from Postgres and from tree (RocksDB).
+It is possible to configure a ZKsync node to periodically prune all data from L1 batches older than a configurable
+threshold. Data is pruned both from Postgres and from tree (RocksDB). Pruning happens continuously (i.e., does not
+require stopping the node) in the background during normal node operation. It generally does not significantly influence
```
Comment (tomg10) [inline docs/guides/external-node/08_pruning.md:5]: I'd rephrase this to something like "It's designed to generally not influence node performance"

**Review by tomg10 — DISMISSED** [review]
Good work! I left two minor comments

File: docs/guides/external-node/08_pruning.md (hunk 1 above)
Comment (slowli) [reply docs/guides/external-node/08_pruning.md:8]: I think it makes sense to list main kinds of data that is getting pruned, because readers may not have good intuition regarding this.

Comment (tomg10) [reply docs/guides/external-node/08_pruning.md:8]: then at least write a TLDR of what is pruned, my first reaction to this list was something like "there are some kinds of data that are pruned and some that aren't" We'd also decrease the risk for this list to get out-of-date

**Review by tomg10 — APPROVED** [review]
You can probably dismiss my other comment, after longer thought it's fine as it is!
